src/LTL/Algorithm/NestedDepthFirstSearch.cpp

```diff
@@ -58,19 +61,18 @@ namespace LTL {
     template<typename T>
     void NestedDepthFirstSearch::dfs(ProductSuccessorGenerator<T>& successor_generator)
     {
-        this->_is_weak = successor_generator.is_weak() && this->_shortcircuitweak;
```

Comment:
This deletion is overly quick, since `ProductSuccessorGenerator::is_weak()` asks the underlying Buchi automaton whether it is weak and thus whether the optimization is sound. Reducing the number of flags involved is good, but the successor generator check is important.

Also, currently the `LTLSearch` call in main incorrectly passes the seed offset to the weak flag (since there are default arguments on both of those).

Reply:
For NDFS this is superseeded by the check in:
```c++
                if (is_new) {
                    if(_shortcircuitweak &&
                       successor_generator.is_accepting(curState) &&
                       successor_generator.has_invariant_self_loop(curState))
                    {
                        _violation = true;
                        if(_build_trace)
                            build_trace(todo, nested_todo);
                        return;
                    }
                    todo.push_back(stack_entry_t<T>{stateid, T::initial_suc_info()});
                }
 ```
 where the weakness of a single buchi-state is checked.

Reply:
lines 107-119 of NestedDepthFirstSearch.cpp

Reply:
1) This is a different check to Büchi weakness though - a weak automaton is one where SCCs either contain no accepting states or only accepting states, so any loop there would allow for immediate termination without a need for the nested step. Checking whether we are in an accepting state with a true self loop is a different case which is also independent from whether the automaton is weak.

2) The change on those lines only weakens the check that was in place previously, since now the weak flag unnecessarily impacts that optimization.

3) The flag still does not appear to relate to whether the Büchi automaton is actually weak, only to whether the user specified `-noweak` on the command-line

Reply:
Probably also worth mentioning is that we don't seem to lose points.